test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/drools/DroolsSessionFailoverIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+        smartRouterRuleServiceClient = smartRouterServicesClient.getServicesClient(RuleServicesClient.class);
+
+        kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+
+        deploymentScenario.getSmartRouterDeployment().setRouterTimeout(SMART_ROUTER_OPENSHIFT_ROUTER_TIMEOUT);
+    }
+
+    @Test
+    public void executeSimpleRuleFailoverTest() throws InterruptedException {
+        logger.debug("Register Kie Container to Kie Server");
+        KieServerInfo serverInfo = kieServerClient.getServerInfo().getResult();
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, RULE_PROJECT_NAME, RULE_PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        logger.debug("Get Kie Server Instance");
+        Instance kieServerInstance = deploymentScenario.getKieServerDeployment().getInstances().iterator().next();
+
+        logger.debug("Set Batch command");
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        BatchExecutionCommand batchExecutionCommand = commandsFactory.newBatchExecution(commands, KIE_SESSION);
+        commands.add(commandsFactory.newSetGlobal(LIST_NAME, new ArrayList<String>(), LIST_OUTPUT_NAME));
+        commands.add(commandsFactory.newFireAllRules());
+        commands.add(commandsFactory.newGetGlobal(LIST_NAME, LIST_OUTPUT_NAME));
+
+        logger.debug("Start new thread for Kie server fail");
+        Thread failoverThread = kieServerFailoverThread(kieServerInstance);
+        failoverThread.start();
+
+        logger.debug("Fire all rules");
+        ServiceResponse<ExecutionResults> response;
+        try {
+            response = smartRouterRuleServiceClient.executeCommandsWithResults(CONTAINER_ID, batchExecutionCommand);
+        } catch (Exception e) {
+            failoverThread.join();
+            throw e;
+        }
+
+        logger.debug("Check result of the drools command");
+        KieServerAssert.assertSuccess(response);
+        ExecutionResults result = response.getResult();
+        assertThat(result).isNotNull();
+        List<String> outcome = (List<String>) result.getValue(LIST_OUTPUT_NAME);
+        assertThat(outcome).isNotNull().hasSize(2);
+
+        assertThat(outcome.get(0)).isNotNull().startsWith("Fast first rule in first activation group executed");
+        assertThat(outcome.get(1)).isNotNull().startsWith("Slow rule without activation group executed");
+
+        logger.debug("Check Kie server deployment");
+        assertThat(deploymentScenario.getKieServerDeployment().getInstances()).doesNotContain(kieServerInstance);
```

Comment:
Test should also check that rule was executed on one of currently available nodes. That can be achieved by adding System.out.println(...) to the executed rule.